src/course-outline/configure-modal/ConfigureModal.jsx

```diff
@@ -16,33 +16,113 @@ import { getCurrentItem } from '../data/selectors';
   const [saveButtonDisabled, setSaveButtonDisabled] = useState(true);
+  const [graderType, setGraderType] = useState('Not Graded');
+  const [dueDateState, setDueDateState] = useState('');
+  const [isTimeLimitedState, setIsTimeLimitedState] = useState(false);
+  const [defaultTimeLimitMin, setDefaultTimeLimitMin] = useState(30);
+  const [hideAfterDueState, setHideAfterDueState] = useState(false);
+  const [showCorrectnessState, setShowCorrectnessState] = useState(false);
 
   useEffect(() => {
     setReleaseDate(sectionStartDate);
   }, [sectionStartDate]);
 
+  useEffect(() => {
+    setDueDateState(dueDate);
+  }, [dueDate]);
+
+  useEffect(() => {
+    setIsTimeLimitedState(isTimeLimited);
+  }, [isTimeLimited]);
+
+  useEffect(() => {
+    setDefaultTimeLimitMin(defaultTimeLimitMinutes);
+  }, [defaultTimeLimitMinutes]);
+
+  useEffect(() => {
+    setHideAfterDueState(hideAfterDue);
+  }, [hideAfterDue]);
+
+  useEffect(() => {
+    setShowCorrectnessState(showCorrectness);
+  }, [showCorrectness]);
+
   useEffect(() => {
     setIsVisibleToStaffOnly(visibilityState === VisibilityTypes.STAFF_ONLY);
   }, [visibilityState]);
 
   useEffect(() => {
     const visibilityUnchanged = isVisibleToStaffOnly === (visibilityState === VisibilityTypes.STAFF_ONLY);
-    setSaveButtonDisabled(visibilityUnchanged && releaseDate === sectionStartDate);
-  }, [releaseDate, isVisibleToStaffOnly]);
+    setSaveButtonDisabled(
+      visibilityUnchanged
+      && releaseDate === sectionStartDate
+      && dueDateState === dueDate
+      && isTimeLimitedState === isTimeLimited
+      && defaultTimeLimitMin === defaultTimeLimitMinutes
+      && hideAfterDueState === hideAfterDue
+      && showCorrectnessState === showCorrectness
+      && graderType === 'Not Graded',
+    );
+  }, [
+    releaseDate,
+    isVisibleToStaffOnly,
+    dueDateState,
+    isTimeLimitedState,
+    defaultTimeLimitMin,
+    hideAfterDueState,
+    showCorrectnessState,
+    graderType,
+  ]);
 
   const handleSave = () => {
-    onConfigureSubmit(isVisibleToStaffOnly, releaseDate);
+    if (isSubsection) {
+      onConfigureSubmit(
+        isVisibleToStaffOnly,
+        releaseDate,
+        graderType === 'Not Graded' ? 'notgraded' : graderType,
+        dueDateState,
+        isTimeLimitedState,
+        defaultTimeLimitMin,
+        hideAfterDueState,
+        showCorrectnessState,
+      );
+    } else {
+      onConfigureSubmit(isVisibleToStaffOnly, releaseDate);
+    }
+  };
+
+  const handleClose = () => {
+    setReleaseDate(sectionStartDate);
+    setDueDateState(dueDate);
+    setIsTimeLimitedState(isTimeLimited);
+    setDefaultTimeLimitMin(defaultTimeLimitMinutes);
+    setHideAfterDueState(hideAfterDue);
+    setShowCorrectnessState(showCorrectness);
+    setIsVisibleToStaffOnly(visibilityState === VisibilityTypes.STAFF_ONLY);
+    setGraderType('Not Graded');
```

Comment:
Same here. We should not reset to `Not Graded` by default, value from the state should be used.